### DISCUSSION TITLE
[Question] Why should anyone consider using Store if they are already using NetworkBoundResource concept/framework?

### DISCUSSION THREAD

**Discussion by RohitSurwase** [dropbox/Store#596] [Q&A]
Today, I watched the [Network-Resilient Applications with Store5 | Talking Kotlin [#128]](https://www.youtube.com/watch?v=a32Otwx7c0w) video, recently published. And the immediate question that I got is, why should anyone use Store? We already have NetworkBoundResource concept/framework which does similar to what Store does with very simple to understand and minimal code. We can alter it the way we want (to make it active or passive or use it in RxJava or Coroutines) and it works perfect.

Is Store much more than that? If YES, why do we need that additional part? We can keep aside its multi-platform support aside for now as it is not yet recommended for production and the project was started for Kotlin initially.

Someone, please explain.

**Comment by digitalbuddha** [comment]
if you don't need the additional functionality then keep doing what you do. As for why others use store:
Request/response types 
Memory caching 
Thousands of tests 
Factories for creating stores from various fetcher and source of truth types. 
Validators
There's many more features you can find in the docs but no one can really say if you need them, that's entirely up to you ♥️
